aten/src/TH/THMath.h

```diff
@@ -129,34 +129,123 @@ Date:  February 1996
 }
 #undef CENTRAL_RANGE
 
-static inline double TH_digamma(double x) {
+static inline double TH_polevl(double x, double *A, size_t len) {
   double result = 0;
-  if (x < 0.5) {
-    result -= M_PI / tan(M_PI * x);
-    x = 1 - x;
+  for (size_t i = 0; i <= len; i++) {
+    result = result * x + A[i];
+  }
+  return result;
+}
+
+static inline float TH_polevlf(float x, float *A, size_t len) {
```

Comment:
Can we use templates to deduplicate those?

Reply:
I think this is a header file for a C file so no, I don't think we can use templates here :(

Reply:
I think we can make it .cpp, right?

Reply:
I think I'd have to convert THTensorMath.c into a cpp file or make a new one for polevl. I'll leave this for now; there's a lot of repeated code in THMath.h that can benefit from a future cleanup like this.